services/notifications/pkg/email/email.go

```diff
@@ -17,49 +20,74 @@ var (
 )
 
 // RenderEmailTemplate renders the email template for a new share
-func RenderEmailTemplate(mt MessageTemplate, locale string, emailTemplatePath string, translationPath string, vars map[string]interface{}) (string, string, error) {
-	// translate a message
-	mt.Subject = ComposeMessage(mt.Subject, locale, translationPath)
-	mt.Greeting = ComposeMessage(mt.Greeting, locale, translationPath)
-	mt.MessageBody = ComposeMessage(mt.MessageBody, locale, translationPath)
-	mt.CallToAction = ComposeMessage(mt.CallToAction, locale, translationPath)
+func RenderEmailTemplate(mt MessageTemplate, locale string, emailTemplatePath string, translationPath string, vars map[string]interface{}) (*channels.Message, error) {
+	textMt, err := NewTextTemplate(mt, locale, translationPath, vars)
+	if err != nil {
+		return nil, err
+	}
+	tpl, err := parseTemplate(emailTemplatePath, mt.textTemplate)
+	if err != nil {
+		return nil, err
+	}
+	textBody, err := emailTemplate(tpl, textMt)
+	if err != nil {
+		return nil, err
+	}
 
-	// replace the body email placeholders with the values
-	subject, err := executeRaw(mt.Subject, vars)
+	htmlMt, err := NewHTMLTemplate(mt, locale, translationPath, vars)
 	if err != nil {
-		return "", "", err
+		return nil, err
+	}
+	htmlTpl, err := parseTemplate(emailTemplatePath, filepath.Join("html", "email.html.tmpl"))
+	if err != nil {
+		return nil, err
+	}
+	htmlBody, err := emailTemplate(htmlTpl, htmlMt)
+	if err != nil {
+		return nil, err
 	}
 
-	// replace the body email template placeholders with the translated template
-	rawBody, err := executeEmailTemplate(emailTemplatePath, mt)
+	var data map[string][]byte
+	data, err = readImages(emailTemplatePath)
 	if err != nil {
-		return "", "", err
+		data, err = readFs()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I don't like the try/catch logic. Two possible alternatives:

(1) Simple Solution: check `emailTemplatePath == ""`  and use `readFS` if true, otherwise `readImages`

(2) Master Solution: Make `RenderEmailTemplate` use an `fs.FS` instead the `emailTemplatePath`. On service start create a `fs.FS` either using `templatesFS` or `os.DirFS(emailTemplatePath)`

Reply:
Sure, I'll change it. I just followed the current approach https://github.com/owncloud/ocis/blob/master/services/notifications/pkg/email/email.go#L49

Reply:
That's not nice either :smile: Let's do it correctly here.

Reply:
done

Reply:
The NOTIFICATIONS_EMAIL_TEMPLATE_PATH will be like this
```
{NOTIFICATIONS_EMAIL_TEMPLATE_PATH}/templates/text/email.text.tmpl
{NOTIFICATIONS_EMAIL_TEMPLATE_PATH}/templates/html/email.html.tmpl
```
@micbar @kobergj @dragonchaser are you ok with it?
I updated the readme